src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -510,56 +526,78 @@ private int getPossibleNewSlavesCount(SlaveTemplate template) throws AmazonClien
-                provisionOptions = EnumSet.of(SlaveTemplate.ProvisionOptions.FORCE_CREATE);
-            else if (possibleSlavesCount > 0)
+        EC2AbstractSlave ret = null;
+//        List<Node> canceledNodes = new ArrayList<Node>();
+
+        // workaround for JENKINS-37483, don't mess with the Jenkins node list while holding a lock
+
+//            int possibleSlavesCount = getPossibleNewSlavesCount(template, canceledNodes);
+//           canceledNodes if (possibleSlavesCount < 0) {
+//                LOGGER.log(Level.INFO, "Cannot provision - no capacity for instances: " + possibleSlavesCount);
+//                return null;
+//            }
+
+            try {
+                EnumSet<SlaveTemplate.ProvisionOptions> provisionOptions = EnumSet.noneOf(SlaveTemplate.ProvisionOptions.class);
                 provisionOptions = EnumSet.of(SlaveTemplate.ProvisionOptions.ALLOW_CREATE);
-            return template.provision(StreamTaskListener.fromStdout(), requiredLabel, provisionOptions);
-        } catch (IOException e) {
-            LOGGER.log(Level.WARNING, "Exception during provisioning", e);
-            return null;
-        }
+                if (forceCreateNew)
+                    provisionOptions = EnumSet.of(SlaveTemplate.ProvisionOptions.FORCE_CREATE);
+
+                ret = template.provision(StreamTaskListener.fromStdout(), requiredLabel, provisionOptions);
+            } catch (IOException e) {
+                LOGGER.log(Level.WARNING, "Exception during provisioning", e);
+                ret = null;
+            }
+
+
+//        for (Node n : canceledNodes) {
+//            try {
+//                Jenkins.getInstance().removeNode(n);
+//            } catch (Exception e) {
+//                LOGGER.log(Level.INFO, "Failed to remove canceled node: " + n, e);
+//            }
+//        }
+
+        return ret;
     }
 
     @Override
-    public Collection<PlannedNode> provision(Label label, int excessWorkload) {
+    public Collection<PlannedNode> provision(final Label label, int excessWorkload) {
         try {
-            List<PlannedNode> r = new ArrayList<PlannedNode>();
+            List<PlannedNode> r = new ArrayList<>();
             final SlaveTemplate t = getTemplate(label);
             LOGGER.log(Level.INFO, "Attempting to provision slave from template " + t + " needed by excess workload of " + excessWorkload + " units of label '" + label + "'");
             if (label == null) {
                 LOGGER.log(Level.WARNING, String.format("Label is null - can't calculate how many executors slave will have. Using %s number of executors", t.getNumExecutors()));
             }
             while (excessWorkload > 0) {
-                final EC2AbstractSlave slave = getNewOrExistingAvailableSlave(t, label, false);
+
                 // Returned null if a new node could not be created
-                if (slave == null)
-                    break;
+//                if (slave == null)
+//                    break;
```

Comment:
Moving slave provision via calling `getNewOrExistingAvailableSlave` within the anonymous call method when adding `PlannedNode`.